lemmatization/services/chinese.py

```diff
@@ -0,0 +1,223 @@
+import re
+from typing import Iterable, List, Tuple  # can use `list` and `tuple` in python 3.9
+import unicodedata
+
+from LAC import LAC
+
+from .base import Service
+from ..tokenizer import Tokenizer
+
+TripleIterable = Iterable[Tuple[str, str, str]]
+
+
+class ChineseService(Service):
+    """
+    Lemmatizer Service for Chinese (zho) text, both simplified and traditional forms.
+    """
+
+    SID = "chinese"  # for both simplified and traditional forms
+    LANGUAGES = ["zho"]
+
+    def lemmatize(self, form: str) -> List[str]:
+        """
+        Lemmatized forms are just the words themselves, assuming they've been
+        tokenized by a Chinese segmentation parser.
+        """
+        return [form]
+
+
+class ChineseTokenizer(Tokenizer):
+    """
+    Tokenizer for Mandarin Chinese (Zhongwen).
+    """
+
+    def tokenize(self, text: str) -> TripleIterable:
+        """
+        Tokenize using a standard python Chinese parser.
+        """
+        # 1. normalize text stream (convert some equivalent
+        # characters to their standard forms)
+        unicode_normalized_text = normalize_chinese(text)
+
+        # 2. perform any text transforms to ensure high-quality
+        # segmentation
+        prepared_text = self.prepare_for_segmentation(unicode_normalized_text)
+
+        # 3. perform Chinese segmentation
+        tokens = self.tokenize_chinese_text(prepared_text)
+
+        # 4. Apply post-processing to the token stream to ensure
+        # that lattice lookups will be optimal
+        tokens = self.re_tokenize(tokens)
+
+        # 5. Generate stream of lemmatized tokens
+        lemma_triples = get_triples(tokens)
+
+        return lemma_triples
+
+    def prepare_for_segmentation(self, text: str) -> str:
+        """ Override this method """
```

Comment:
Does this docstring imply that these methods must be overridden in order for things to work properly? (not sure if that sentence makes sense)

If so, what do you think about `raise NotImplementedError` to reinforce that idea. [Link to docs](https://docs.python.org/3/library/exceptions.html#NotImplementedError)

Reply:
A better idea for sure! Thanks! Done in c791be7.